### DISCUSSION TITLE
Permission filter rules on User

### DISCUSSION THREAD

**Discussion by NecoHorne** [directus/directus#9847] [Archive]
Hi

I have a use case where users in a specific role need to be able to view items assigned to them. 

Easily sorted by using: 

```
    {
      "assigned_to": {
        "id": {
          "_eq": "$CURRENT_USER"
        }
      }
    }
```

This rule works as expected.

Where I am struggling is that within that same role there are team leaders/managers who should also be able to see the items assigned to their team members. 

I added a one to many field in the directus_users table called manages that relates back to the directus_users table.
![image](https://user-images.githubusercontent.com/30078491/141992674-2432127d-e924-4afd-b319-4bda8e840f7a.png)

Now if I assign users in this field it all works well, I can see it in the api as expected:
![image](https://user-images.githubusercontent.com/30078491/141992967-a2efd139-e99a-41e2-bae2-6cfb8ca78880.png)

Then in the permissions I create this rule using the `_or` operator to check if it is assigned to the logged in user or if the item is assigned to a user being managed by the signed in user using the `_in` operator

```
{
  "_or": [
    {
      "assigned_to": {
        "id": {
          "_eq": "$CURRENT_USER"
        }
      }
    },
    {
      "assigned_to": {
        "id": {
          "_in": "$CURRENT_USER.manages"
        }
      }
    }
  ]
}
```

The problem is this leads to an error: 

![image](https://user-images.githubusercontent.com/30078491/141993516-6a06e14f-c12d-4e9b-8cf1-6137425364bf.png)

```
select `dealer`.`bluechip_id`, `dealer`.`dealer_name`, `dealer`.`status`, `dealer`.`id`, `dealer`.`assigned_to` from `dealer` left join `directus_users` as `didqy` on `dealer`.`assigned_to` = `didqy`.`id` left join
`directus_users` as `fhlca` on `dealer`.`assigned_to` = `fhlca`.`id` where (not `dealer`.`status` = 'archived' and ((`fhlca`.`id` = '9113e8e4-80c4-4c50-92ce-51176c0d18c1' or `fhlca`.`id` in (('00387735-0040-4d72-99a1-c227ed04170f
', '0f267df6-64c3-4b23-beeb-1cc8e7c1aec9', '25214bbe-4dfa-4e43-9951-95e1449e0e2c', 'a5fcfe76-3fbb-4f33-96d3-851bf903664a', 'dc579dfd-fcdc-4b13-aefc-28cf35af2052', 'f8c52271-81c7-4cbe-b785-6f2268231ced'))))) order by `dealer`.`id`
 asc limit 25 - ER_OPERAND_COLUMNS: Operand should contain 1 column(s) 
```

Am I doing something wrong with my filtering rules or is this a bug? I can see in the error message it is trying the in array method but something seems off. 

EDIT:

Using the following by itself works as expected too. So the issue does seem to be in my nesting of the permissions

```
 {
      "assigned_to": {
        "id": {
          "_in": "$CURRENT_USER.manages"
        }
      }
    }
```

more info: 
currently on `v9.0.1`
database is Amazon MySQL RDS instance.

**Comment by NecoHorne** [comment]
Managed to solve this using the following filter: 
```
{
	"assigned_to": {
		"id": {
			"_in": [
				"$CURRENT_USER",
				"$CURRENT_USER.manages"
			]
		}
	}
}
```

Still not 100% why the `_or` does not work for this but this will work for my purposes for now.

**Comment by rijkvanzanten** [comment]
Your original example looks good to me and should work as expected. Mind opening a new issue if the problem still persists?
